main.go

```diff
@@ -397,143 +395,88 @@ func mainCore() error {
-	// since when syncing page was running no other page that can be displayed
-	// thus no monitorss that were running.
-	collectBlocksAfterSyncing := func() error {
-		// Blockchain monitor for the collector
-		addrMap := make(map[string]txhelpers.TxAction) // for support of watched addresses
-		// On reorg, only update web UI since dcrsqlite's own reorg handler will
-		// deal with patching up the block info database.
-		reorgBlockDataSavers := []blockdata.BlockDataSaver{explore}
-		wsChainMonitor := blockdata.NewChainMonitor(collector, blockDataSavers,
-			reorgBlockDataSavers, quit, &wg, addrMap,
-			notify.NtfnChans.ConnectChan, notify.NtfnChans.RecvTxBlockChan,
-			notify.NtfnChans.ReorgChanBlockData)
+	// latestBlockHash receives the block hash of the latest block to be sync'd
+	// in dcrdata. This may not necessarily be the latest block in the blockchain
+	// but it is the latest block to be sync'd according to dcrdata. This block
+	// hash is sent when blockchain syncing is happening simulatneoulsy as the
+	// running webserver servers the normal pages on initial dcrdata startup.
+	latestBlockHash := make(chan *chainhash.Hash)
 
-		// Blockchain monitor for the stake DB
-		sdbChainMonitor := baseDB.NewStakeDBChainMonitor(quit, &wg,
-			notify.NtfnChans.ConnectChanStakeDB, notify.NtfnChans.ReorgChanStakeDB)
-
-		// Blockchain monitor for the wired sqlite DB
-		wiredDBChainMonitor := baseDB.NewChainMonitor(collector, quit, &wg,
-			notify.NtfnChans.ConnectChanWiredDB, notify.NtfnChans.ReorgChanWiredDB)
+	// Initiate the sync status monitor if the sync status is activated else
+	// initiate the goroutine that handle storing blocks needed for the explorer
+	// pages
+	if explore.DisplaySyncStatusPage {
+		explore.StartSyncingStatusMonitor()
 
-		var auxDBChainMonitor *dcrpg.ChainMonitor
-		auxDBBlockConnectedSync := func(*chainhash.Hash) {}
-		if usePG {
-			// Blockchain monitor for the aux (PG) DB
-			auxDBChainMonitor = auxDB.NewChainMonitor(quit, &wg,
-				notify.NtfnChans.ConnectChanDcrpgDB, notify.NtfnChans.ReorgChanDcrpgDB)
-			if auxDBChainMonitor == nil {
-				return fmt.Errorf("Failed to enable dcrpg ChainMonitor. *ChainDB is nil.")
+	} else {
+		// set that blocks freshly sync'd to to be stored for the explorer pages
+		// till the sync is done
+		explorer.SetSyncExplorerUpdateStatus(true)
+
+		// This goroutines updates the blocks needed on the explorer pages. It
+		// only runs when the status sync page is not the defualt page that a user
+		// can view on the running webserver but the syncing of blocks behind the
+		// best block is happening in the background. No new blocks monitor from
+		// dcrd will be initiated till the best block from dcrd is in sync with
+		// the best block from dcrdata.
+		go func() {
+			// After the sync on initial startup is complete this goroutine is no
+			// longer needed, so kill it.
+			defer runtime.Goexit()
+
+			isStatusSyncingDisabled := make(chan bool)
```

Comment:
Is this just to get out of the `for` loop below?  If so, I do not think it is working as intended.

In any case, to get out of the outer loop, not just the select, you can tag the for and specify that in the break.  e.g.:

```go
out:
	for {
		select {
		case ...:
		if ... {
			break out
...
```

However, the select is not needed since with just one useful case, you can use for range over a channel:

```go
for hash := range latestBlockHash {
...
```

See https://golang.org/ref/spec#For_range

Reply:
```golang
for hash := range latestBlockHash {
...
```
This version isn't working.  I will revert to using the labels